### PR TITLE
[EMB-119] Add feedback button Ember component

### PR DESCRIPTION
**Pull request by sloria**
## Purpose

Add feedback button to Quick Files page

osf steven loria s quick files 2018-02-08 16-33-39

osf steven loria s quick files 2018-02-08 16-34-14

osf steven loria s quick files 2018-02-08 16-35-17

Example JIRA issue: https://openscience.atlassian.net/browse/FEED-9

This replaces #67 

## Summary of Changes

- Add feedback-button component

## Side Effects / Testing Notes

- Requires the following configuration (for staging and prod):

```js
       microfeedback: {
            url: 'https://microfeedback-osf.herokuapp.com/13209/3',
            pageParams: {
                QuickFiles: {
                    componentID: '13836',
                    priorityID: '10100',
                },
            },
        },
```

## Ticket

https://openscience.atlassian.net/browse/EMB-119

# Reviewer Checklist

- [ ] meets requirements
- [ ] easy to understand
- [ ] DRY
- [ ] testable and includes test(s)
- [ ] changes described in `CHANGELOG.md`

close #67 
close #71

### CODE REVIEW

File: config/environment.js [hunk 1]
```diff
@@ -46,6 +46,17 @@ module.exports = function(environment) {
             },
         ],
         FB_APP_ID: process.env.FB_APP_ID,
+        microfeedback: {
+            url: process.env.MICROFEEDBACK_URL,
```
Comment (icereval) [inline config/environment.js]: This is not needed, send the desired config over to @binoculars, we can inject it at deployment time.

File: config/environment.js [hunk 2]
```diff
@@ -46,6 +46,17 @@ module.exports = function(environment) {
...
+            JIRA: {
+                components: {
+                    QuickFiles: '13836',
```
Comment (icereval) [inline config/environment.js]: This is not needed, send the desired config over to @binoculars, we can inject it at deployment time.

Comment (sloria) [reply config/environment.js]: I guess the idea was to reduce the required configuration. Only `config.microfeedback.url` needs to be configured.

If it's not configured, i.e. `config.microfeedback.url` is `undefined`, the JIRA information here won't be used.

But if you think this doesn't belong, I can remove it.

Comment (binoculars) [reply config/environment.js]: What is the `'13836'` from? It looks like it's specific to our JIRA. I'd like to keep the configurations as generic as possible.

Generally, I like to use environment variables for local development, but I can override the ember environment at deploy time. Maybe something like `microfeedback.options`

File: app/user-quickfiles/controller.js [hunk 3]
```diff
@@ -24,4 +25,6 @@ export default Controller.extend(Analytics, {
             }
         },
     },
+
+    jiraComponent: config.microfeedback.JIRA.components.QuickFiles,
```
Comment (binoculars) [inline app/user-quickfiles/controller.js]: jiraComponent is very environment-specific. Maybe something like `microfeedbackOptions`

**Comment by sloria** [comment]
@binoculars I've removed the JIRA IDs from the config.

I'm not sure there's a whole lot of value in a standalone ember addon. It would have to know how dialogs are displayed--for example, this PR is tied to the fact that we are using ember-bootstrap. So really it would be `ember-bootstrap-microfeedback-addon`--awfully specific, yeah?

I'm not entirely opposed to having a standalone addon, but can be a future improvement. Even if we did that, we would still want to have the `feedback-component` in ember-osf-web that had OSF-specific stuff, e.g., the "Contact me for further opportunities to improve the OSF" checkbox, etc.

**Comment by sloria** [comment]
Discussed with @binoculars : JIRA-specific config is now removed. The ember config will contain the page-specific query params. This allows different backends to be used (not just microfeedback-jira).

**Comment by sloria** [comment]
With the way this PR works, this is the necessary configuration:

```js
        microfeedback: {
            url: 'https://microfeedback-osf.herokuapp.com/13209/3',
            pageParams: {
                QuickFiles: {
                    componentID: '13836',
                    priorityID: '10100',
                }
            },
        },
```

This can be the same for all deployments, incl. production. @binoculars

**Comment by brianjgeiger** [comment]
@binoculars Is this good enough for the 0.2.0 release?

**Comment by binoculars** [comment]
I don't see why not. Should be a quick QA once merged and the configs are updated.

**Comment by brianjgeiger** [comment]
Cool. Can you do an Approval review to keep it all official?

**Pull request merged**